src/plugins/world_control/WorldControl.cc

```diff
@@ -238,33 +274,47 @@ void WorldControl::OnWorldStatsMsg(const ignition::msgs::WorldStatistics &_msg)
 /////////////////////////////////////////////////
 void WorldControl::OnPlay()
 {
-  std::function<void(const ignition::msgs::Boolean &, const bool)> cb =
-      [this](const ignition::msgs::Boolean &/*_rep*/, const bool _result)
-  {
-    if (_result)
-      QMetaObject::invokeMethod(this, "playing");
-  };
-
-  ignition::msgs::WorldControl req;
-  req.set_pause(false);
+  ignition::msgs::WorldControl msg;
+  msg.set_pause(false);
   this->dataPtr->pause = false;
-  this->dataPtr->node.Request(this->dataPtr->controlService, req, cb);
+  if (this->dataPtr->useEvent)
```

Comment:
This block from 280 to 294 is repeated 3 times in different functions, consider moving it to a separate function to reduce duplication

Reply:
Done in 8e7cb8f